File: src/subside/plugins/koth/scheduler/MapRotation.java
package subside.plugins.koth.scheduler;

import java.util.List;

import org.json.simple.JSONObject;

import lombok.Getter;
import lombok.Setter;
import subside.plugins.koth.utils.JSONSerializable;

public class MapRotation implements JSONSerializable<MapRotation> {
    
    public List<String> rotation;
    public @Getter @Setter int index;
    
    public MapRotation(List<String> rotation){
        this.rotation = rotation;
    }
    
    public String getNext(){
        return this.rotation.get(index++%rotation.size());
    }

    @Override
    public MapRotation load(JSONObject obj) {
        this.index = (int)(long)obj.get("index");
        
        return this;
    }

    @SuppressWarnings("unchecked")
    @Override
    public JSONObject save() {
        JSONObject obj = new JSONObject();
        obj.put("index", this.index);
        
        return obj;
    }
    
}

File: src/subside/plugins/koth/scheduler/MapRotation.kt
package subside.plugins.koth.scheduler

import org.bukkit.configuration.serialization.ConfigurationSerializable
import org.bukkit.inventory.ItemStack
import subside.plugins.koth.loot.Loot

class MapRotation(
    val rotation: List<String> = ArrayList(),
    var index: Int = 0
)/* : ConfigurationSerializable*/ {
//    override fun serialize(): Map<String, Any> {
//        return mapOf(
//            "rotation" to rotation,
//            "index" to index
//        )
//    }

    fun getNext(): String {
        return rotation[index++ % rotation.size]
    }

//    companion object {
//        @JvmStatic fun deserialize(args: Map<String, Any>): MapRotation {
//            return MapRotation(
//                args["rotation"] as List<String>,
//                args["index"] as Int
//            )
//        }
//    }
}